### DISCUSSION TITLE
Auto complete and `explainError`

### DISCUSSION THREAD

**Discussion by bugb** [mrcjkb/rustaceanvim#166] [Q&A] [answered]
### Neovim version (nvim -v)

v0.9.4

### Operating system/version

Arch Linux

### Output of :checkhealth rustaceanvim

```console
==============================================================================
rustaceanvim: require("rustaceanvim.health").check()

Checking for Lua dependencies ~
- WARNING dap not installed. Needed for debugging features [mfussenegger/nvim-dap](https://github.com/mfussenegger/nvim-dap)

Checking external dependencies ~
- OK rust-analyzer: found rust-analyzer 1 (21b06c1be 2023-12-15)
- OK Cargo: found cargo 1.65.0 (4bc8f24d3 2022-10-20)
- OK rustc: found rustc 1.65.0 (897e37553 2022-11-02)

Checking config ~
- OK No errors found in config.

Checking for conflicting plugins ~
- OK No conflicting plugins detected.
```

### How to reproduce the issue

When open any rust file I did not see the autosuggestion.

Here is the example:

```bash
use
fn main() {
    println!("Hello, world!");
}
```

Also command like `RustLsp explainError` returns

```
No explainnable errors found
```
![image](https://github.com/mrcjkb/rustaceanvim/assets/9280412/91017a61-2c05-412e-ac51-ac3d93526246)

This is the first time I install plugins like `rustaceanvim` so any help would be appreciated, maybe I missed something for the setup.

### Expected behaviour

The plugin should work!

### Actual behaviour

It does not work!

### The minimal config used to reproduce this issue.

As you can see in the healthcheck section: I have install `neovim` version > `0.9`, `rust`, `rust-analyzer`, `rustaceanvim`.
Here is [the everything that changed](https://github.com/bugb/dotfiles/commit/9f114ef9cf710d25d3cccc7193e05e132b7105dc) for my neovim config file.

**Comment by mrcjkb** [comment]
Hey 👋 

Moving up Q&A discussions, as this is highly unlikely to be a bug.

**Comment by mrcjkb** [comment, marked as answer]
### Auto completions

This plugin doesn't add support for auto completions to Neovim.

You have a few choices for that:

- Use Neovim's built-in omni completion, by typing `<C-X><C-O>` (ctrl+x, then ctrl+o) in insert mode, and using `<C-N>` and `<C-P>` to navigate.
  This supports basic LSP completions, but not autocompletion.
- For autocompletion, you need a plugin, for example [nvim-cmp](https://github.com/hrsh7th/nvim-cmp) + an [LSP completion source](https://github.com/hrsh7th/cmp-nvim-lsp).

### `explainError`

This is the expected behaviour. `explainError` looks for error diagnostics **that have an error code**.
If there are none, it will tell you that there were `no explainable errors found`.

**Comment by bugb** [comment]
Thank @mrcjkb for the detailed explaination, I have made it work by run `cargo new [project]` without install any autocomple/suggestion plugins.
![image](https://github.com/mrcjkb/rustaceanvim/assets/9280412/0a2ac5f5-cb22-496a-a726-5ad52aaf5dca)

**Reply by mrcjkb** [reply]
I see, you were in a standalone file...
The latest version of this plugin should work with that, too.

**Comment by afifurrohman-id** [comment]
> ### Auto completions
> This plugin doesn't add support for auto completions to Neovim.
> 
> You have a few choices for that:
> 
> * Use Neovim's built-in omni completion, by typing `<C-X><C-O>` (ctrl+x, then ctrl+o) in insert mode, and using `<C-N>` and `<C-P>` to navigate.
>   This supports basic LSP completions, but not autocompletion.
> * For autocompletion, you need a plugin, for example [nvim-cmp](https://github.com/hrsh7th/nvim-cmp) + an [LSP completion source](https://github.com/hrsh7th/cmp-nvim-lsp).
> 
> ### `explainError`
> This is the expected behaviour. `explainError` looks for error diagnostics **that have an error code**. If there are none, it will tell you that there were `no explainable errors found`.

Ah finally i found this helpful discussion after a few hours search why i cannot get auto completion :(.

Could you please also add note in `README.md` it will helpful for someone first started.
